### ISSUE TITLE
generate invoice document

### ISSUE THREAD

**Issue by bburdette** [InterstitialTech/timeclonk#28]
would be great to automatically generate an invoice document.  

various ideas:

- sum the number of hours, OR
- present all the hours logged summed with the task strings.
- use project settings to fill in the blanks on the invoice.
- store client information in the project record - address, name, etc.  
- upload a template docx document with text that is replaced by the above.  

there is a library, docx-rs, that might be able to help with this.  currently is a bit broken but the examples seemed to work after fixing the compile errors.

**Comment by bburdette** [comment]
Josh suggested using latex for doing an invoice template.  Looks better for templating than docx.

**Comment by bburdette** [comment]
Some latex invoice templates:

http://www.latextemplates.com/cat/invoices

**Comment by bburdette** [comment]
@muffinator what would be great would be a way to search-and-replace fields in the invoice.  I could supply the fields in json form, something like

```
[ { "search": "{client address}"
   "replace": "1234 5th Street" },
  { "search": "{total hours}"
   "replace": "24.5" },
  {"search": "{hourly rate}",
   "replace": "85"}
]

Or maybe there's a better format that latex prefers for templates?  

Basically if there's a script that takes a table ot search-replace pairs and ewrites an invoice document, I can incorporate that into the server. 
```

**Comment by bburdette** [comment]
https://github.com/typst/typst
